### PR TITLE
Wbhart update cxxwrap

### PR DESCRIPTION
**Pull request by fieker**
should be merged with other pull request - but I lack to git-fu

### CODE REVIEW

**Comment by fieker** [comment]
On Wed, Feb 05, 2020 at 06:00:17AM -0800, Max Horn wrote:
> fingolfin commented on this pull request.
> 
> Seems plausible though before we switch to CxxWrap 0.9, we should
> coordinate with Polymake.jl / @kalmarek to avoid annoyances.
We need to - polymake does not work with 0.9. They also have a patch or
many waiting. It needs to happen at the same time...
> 
> > @@ -15,8 +15,7 @@ BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
>  [compat]
>  AbstractAlgebra = "^0.8.0"
>  Nemo = "^0.16.0"
> -julia = "1.0, 1.1, 1.2"
> -CxxWrap = "^0.8.1"
> +julia = "≥ 1.0.0"
> 
> The commit with this change says "Add dependency on 0.9.0 of CxxWrap" but that's not what happens here. Suggestion:
> ```suggestion
> julia = "≥ 1.0.0"
> CxxWrap = "^0.9"
> ```
Yep. This was an attempt to get it to finally actually use 0.9 - I ran
out of patience about the absolute denial to acknowledge 0.9
> 
> > @@ -203,8 +203,14 @@ push!(Libdl.DL_LOAD_PATH, joinpath(prefixpath, "lib"))
>  
>  cd(oldwdir)
>  
> +println("here")
> 
> ```suggestion
> ```
> 
> >  jlcxx_cmake_dir = realpath(joinpath(dirname(pathof(CxxWrap)), "..", "deps", "usr", "lib", "cmake", "JlCxx"))
>  
> +#jlcxx_cmake_dir = ENV["JLCXX_DIR"]
> +
> +println("jlcxx_cmake_dir = $jlcxx_cmake_dir")
> 
> I assume we can also get rid of this?
Honestly: no idea. I do not understand why this is neccessary. The web
seems to suggest that this should/could be handled different in cmake.
It works.
clearly the prints can be removed.

**Comment by wbhart** [comment]
You should also update the version of Singular to 0.2.0 as this is technically breaking (for Oscar). That would make this independent of the Polymake PR.

File: Project.toml [hunk 1]
```diff
@@ -15,8 +15,7 @@ BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
 [compat]
 AbstractAlgebra = "^0.8.0"
 Nemo = "^0.16.0"
-julia = "1.0, 1.1, 1.2"
-CxxWrap = "^0.8.1"
+julia = "≥ 1.0.0"
```
[the remark this answers is not in the document]
Comment (wbhart) [reply Project.toml]: It's not correct anyway. Only 1.0, 1.1, 1.2, 1.3 and 1.4 should be listed. Putting all versions of Julia greater than a certain one will break things in future.

[the remark this answers is not in the document]
Comment (wbhart) [reply Project.toml]: And the dependency on CxxWrap 0.9 is in the next line below, I believe.

File: deps/build.jl [hunk 2]
```diff
@@ -203,8 +203,14 @@ push!(Libdl.DL_LOAD_PATH, joinpath(prefixpath, "lib"))
 
 cd(oldwdir)
 
+println("here")
```
[the remark this answers is not in the document]
Comment (wbhart) [reply deps/build.jl]: Agreed.

File: deps/build.jl [hunk 3]
```diff
@@ -203,8 +203,14 @@ push!(Libdl.DL_LOAD_PATH, joinpath(prefixpath, "lib"))
...
+
 jlcxx_cmake_dir = realpath(joinpath(dirname(pathof(CxxWrap)), "..", "deps", "usr", "lib", "cmake", "JlCxx"))
 
+#jlcxx_cmake_dir = ENV["JLCXX_DIR"]
+
+println("jlcxx_cmake_dir = $jlcxx_cmake_dir")
```
[the remark this answers is not in the document]
Comment (wbhart) [reply deps/build.jl]: Yes, we can get rid of the commented line and the println. At least, I think so. We should be going back to whatever was there before, I believe, and that would seem to do it.

**Comment by wbhart** [comment]
I don't see where the changes mentioned at the bottom of this thread:

https://github.com/JuliaInterop/CxxWrap.jl/issues/170

have been made.

**Comment by wbhart** [comment]
Also, there are a lot of CI failures which look real.

File: Project.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fingolfin) [reply Project.toml]: It is there because I added it in the meantime :-)

**Comment by kalmarek** [comment]
what is the ETA on this?

**Comment by fieker** [comment]
On Wed, Feb 12, 2020 at 01:04:23PM -0800, kalmarek wrote:
> what is the ETA on this?

no idea - the tests are bizarrly working on MacOS, but not on linux. Its
failing with syntatical errors...

**Comment by kalmarek** [comment]
Linux and MacOS don't share the compiler; 
also there are still two deprecated warnings on MacOS

**Pull request merged**